Rubberduck.Parsing/ComReflection/ComVariant.cs

```diff
@@ -59,12 +88,79 @@ private struct Variant
 
         public ComVariant(IntPtr variant)
         {
-            Value = Marshal.GetObjectForNativeVariant(variant);
             var members = Marshal.PtrToStructure<Variant>(variant);
             VariantType = (VarEnum)members.vt;
-            if (Value == null && VariantType == VarEnum.VT_BSTR)
+
+            // Note that these are technically flags, but it they are VT_BYREF or VT_ARRAY, the data area will always be a pointer.
+            // ReSharper disable once SwitchStatementMissingSomeCases
+            switch ((VarEnum)members.vt)
             {
-                Value = string.Empty;
+                case VarEnum.VT_VOID:
+                case VarEnum.VT_EMPTY:
+                case VarEnum.VT_NULL:
+                case VarEnum.VT_ERROR:
+                    Value = null;
+                    break;
+                case VarEnum.VT_I2:
+                    Value = members.sVal;
+                    break;
+                case VarEnum.VT_HRESULT:
+                case VarEnum.VT_I4:
+                    Value = members.iVal;
+                    break;
+                case VarEnum.VT_R4:
+                    Value = members.fltVal;
+                    break;
+                case VarEnum.VT_R8:
+                    Value = members.dblVal;
+                    break;
+                case VarEnum.VT_CY:
+                    Value = members.dblVal * 10000;
+                    break;
+                case VarEnum.VT_DATE:
+                    Value = DateTime.FromOADate(members.dblVal);
+                    break;
+                case VarEnum.VT_BSTR:
+                    Value = Marshal.PtrToStringBSTR(members.ToPointer());
+                    break;
+                case VarEnum.VT_DISPATCH:
+                    break;
+                case VarEnum.VT_BOOL:
+                    Value = members.boolVal;
+                    break;
+                case VarEnum.VT_I1:
+                    Value = members.sbVal;
+                    break;
+                case VarEnum.VT_UI1:
+                    Value = members.bVal;
+                    break;
+                case VarEnum.VT_UI2:
+                    Value = members.usVal;
+                    break;
+                case VarEnum.VT_UI4:
+                    Value = members.uiVal;
+                    break;
+                case VarEnum.VT_I8:
+                    Value = members.lVal;
+                    break;
+                case VarEnum.VT_UI8:
+                    Value = members.ulVal;
+                    break;
+                case VarEnum.VT_INT:
+                    Value = IntPtr.Size == 4 ? members.iVal : members.lVal;
```

Comment:
Shouldn't you be using `IsProcess32Bit` here? Keep it DRY.

Reply:
Yes. I completely forgot that I'd defined that last night.